engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+
+/**
+ *
+ */
+public class RenderGraph { // TODO: add extends DirectedAcyclicGraph<Node>
+    private List<Node> nodes;
+
+    public RenderGraph() {
+        nodes = Lists.newArrayList();
+    }
+
+    public String add(Node node, String suggestedId) {
```

Comment:
Sort of. If `.add` could be ambiguous about what it is adding then a more detailed name could be nice. However, if the "edge" term is also integral and clear to 3d wizards as somehow distinct from the norm (dealing with nodes is the standard case, dealing with edges is an exception) then that might still work :-)

In other words: I have no idea what I'm talking about as far as 3d wizardry goes, I'm just looking at and commenting about the logic/semantics.